service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/EhrResendControllerTest.java

```diff
@@ -224,6 +231,40 @@ void When_AnEhrExtractDoesNotExist_Expect_RespondsWith404() throws JsonProcessin
         );
     }
 
+    @Test
+    void When_ResendingTransfer_Expect_EhrStatusInProgressStatus() {
+
+        String ehrMessageRef = generateRandomUppercaseUUID();
+        var ehrExtractStatus = new EhrExtractStatus();
+
+        ehrExtractStatus.setConversationId(CONVERSATION_ID);
+        ehrExtractStatus.setEhrReceivedAcknowledgement(EhrExtractStatus.EhrReceivedAcknowledgement.builder()
+                                                           .conversationClosed(FIVE_DAYS_AGO)
+                                                           .errors(List.of(
+                                                               EhrExtractStatus.EhrReceivedAcknowledgement.ErrorDetails.builder()
+                                                                   .code(INCUMBENT_NACK_CODE)
+                                                                   .display(INCUMBENT_NACK_DISPLAY)
+                                                                   .build()))
+                                                           .messageRef(ehrMessageRef)
+                                                           .received(FIVE_DAYS_AGO)
+                                                           .rootId(generateRandomUppercaseUUID())
+                                                           .build());
+        ehrExtractStatus.setEhrRequest(EhrExtractStatus.EhrRequest.builder().nhsNumber(NHS_NUMBER).build());
+        ehrExtractStatus.setEhrExtractCorePending(EhrExtractStatus.EhrExtractCorePending.builder().build());
+        ehrExtractStatus.setEhrContinue(EhrExtractStatus.EhrContinue.builder().build());
+        ehrExtractStatus.setGpcAccessDocument(EhrExtractStatus.GpcAccessDocument.builder().build());
```

Comment:
Is it time to refactor this out into a separate place? I'm pretty sure this is a duplicated in a test above. 🤔